### ISSUE TITLE
Integration with Pyomo (or similar)

### ISSUE THREAD

**Issue by gilirosenberg12** [dwavesystems/dimod#1232]
**Application**
I want to solve my ILP/QP with multiple solvers, including (but not limited) to the QA. I'd much rather use a single piece of code for formulating the problem. For example, via a standard interface like Pyomo. However, I cannot do that for the QA - I need to reformulate the problem using `dimod` (or another library).

**Proposed Solution**
I'm suggesting to add an integration with Pyomo, or a similar library that supports solving non-linear integer optimization problems. I should be able to write code to formulate my optimization problem only once, and solve it with various common conventional solvers like Gurobi, CPLEX, XPRESS, etc., as well as with the QA.

**Alternatives Considered**
Using CQM directly - the existence of CQM makes this relatively easy. However, it introduces redundancy, which can be a source of errors, as well as requiring increased effort to test, validate, and maintain the codebase.

**Comment by arcondello** [comment]
Agree that would be useful. In the mean time, we do support [lp files](https://docs.ocean.dwavesys.com/en/stable/docs_dimod/reference/utilities.html#module-dimod) which can serve as an intermediate representation.

**Comment by gilirosenberg12** [comment]
Thanks, @arcondello I wasn't aware of this option, that should work nicely for now! But still it would be nice to have a more direct integration, to avoid the overhead of writing to a file and reading from it...

**Comment by vpranjal** [comment]
Very much agree with this request - would love to use an integrated approach with Pyomo and pass the problem to QA. CQM might enforce that no cubic or higher order terms be present in the formulation to start with. Might be generally be usable for MILP/ LPs/ QPs/ QCQPs solvable by CQM.

**Comment by matti-sarjala** [comment]
A useful design boundary for a Pyomo integration may be to distinguish “formulated in Pyomo” from “directly representable as a dimod CQM”.

Pyomo can express a broader class of algebraic optimization models than CQM. So a direct integration probably needs an explicit compatibility contract rather than only a file/API bridge.

Possible acceptance criteria:

1. Accept models whose objective and constraints are linear or quadratic.
2. Reject cubic or higher-order terms with a clear error unless an explicit transformation is requested.
3. Preserve variable domains where CQM has matching semantics.
4. Preserve constraint labels or generate stable labels if Pyomo labels are unavailable.
5. Make objective sense and constant offsets explicit.
6. Add tests comparing a small Pyomo model converted through LP and through the direct path, when both are supported.

This would make the integration safer: users can formulate once, but the conversion boundary still tells them when the model is genuinely CQM-compatible and when a reformulation/penalty/linearization step is required.
